### ISSUE TITLE
sha256 hash function on Windows in Godot 3.4 takes close to 10x as long as it does to 3.3

### ISSUE THREAD

**Issue by RickWillcox** [godotengine/godot#55193]
### Godot version

3.4

### System information

Windows10

### Issue description

If I run this code on 3.3 / 3.4 linux or 3.3 windows, I get about 600ms taken to finish `hashit()`. 
If I run the same code (same project even), in 3.4 windows it takes around 5550ms to finish `hashit()`
I have tested this on two machines and also my friend has the same result. We found the bug in our project when we tried to move to 3.4.
```gdscript
func _ready() -> void:
	hashit()

func hashit():
	var start : int = OS.get_system_time_msecs()
	var string : String = "string to hash"
	var ROUNDS : int = pow(2,18)
	var rounds : int = ROUNDS
	while rounds > 0:
		string = string.sha256_text()
		rounds -= 1
	print("sha256 hash run %d times and took %dms" % [ROUNDS, OS.get_system_time_msecs()-start])
```

### Steps to reproduce

run the code on 3.4 windows 
compare to time on 3.3 windows

### Minimal reproduction project

[testhash.zip](https://github.com/godotengine/godot/files/7576264/testhash.zip)

**Comment by Calinou** [comment]
@RickWillcox Can you reproduce this in any of the [3.4 betas and RCs](https://downloads.tuxfamily.org/godotengine/3.4/) to determine when the regression started?

**Comment by RickWillcox** [comment]
We tested it the day 3.4 was released. We are not continuing with this implementation but thought the godot team should know.

**Comment by RickWillcox** [comment]
> 

This issue is only on WINDOWS 3.4. Linux 3.3 /3.4 and WINDOWS 3.3 work fine.

**Comment by akien-mga** [comment]
`String.sha256_text()` calls `CryptoCore::sha256()`, which calls into `mbedtls_sha256_ret`. So this might be related to a mbedtls version update, like https://github.com/ARMmbed/mbedtls/issues/4814. CC @Faless 

@RickWillcox Could you test Godot 3.2.2 and 3.2.3 too? If my hunch is right, then 3.2.3 should be the first version with this regression, 3.2.2 should be fine (mbedtls 2.16.11 and 2.16.10 respectively).

**Comment by RickWillcox** [comment]
I've uploaded the file test file and code. I dont have time to test further.

**Comment by KoBeWi** [comment]
I checked with 3.2.2, 3.2.3 and 3.5 beta5. The MRP took 790-820 ms to execute, with no big differences between versions.

**Comment by akien-mga** [comment]
Assuming fixed, as we updated mbedTLS from 2.16.11 to 2.28.0 between 3.4 and 3.5, which might have solved potential performance regressions in 2.16. Please comment if you can still reproduce the issue so we can investigate further.
